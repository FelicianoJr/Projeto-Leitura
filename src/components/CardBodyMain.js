import React from "react";
import PropTypes from "prop-types";
import { UUID } from "../mocks/UUID";
const CardBodyMain = props => {
  
  const showButton = props.data.commentCount === undefined;

  const newComment = id => {
    return {
      id: UUID(),
      parentId: id,
      timestamp: Date.now(),
    };
  };

  return (
    <div className="card-body">
      {props.data.title && <div className="card-title">{props.data.title}</div>}
      <div className="card-text">{props.data.body}</div>
      <div className="card-text">
        <small>
          <b>Pontuação</b>
        </small>{" "}
        <div className="badge">{props.data.voteScore}</div>
        <br />
        <small className="text-muted">
          {" "}
          Criado por <b>{props.data.author}</b> em <b>{props.data.timestamp}</b>
        </small>
      </div>
      {!showButton && (
        <button
          id="find"
          type="button"
          className="btn btn-outline-info btn-sm mr-sm-2"
          onClick={() => props.onShow(props.data)}
        >
          Visualisar Comentários{" "}
          <span className="badge badge-light">{props.data.commentCount}</span>
        </button>
      )}
      {!showButton && (
        <button
          id="edit"
          type="button"
          className="btn btn-outline-info btn-sm mr-sm-2"
          onClick={() => {
            props.onNew(newComment(props.data.id));
            props.toggleComment();
          }}
        >
          Adicionar Comentário
        </button>
      )}
      <button
        id="edit"
        type="button"
        className="btn btn-outline-info btn-sm mr-sm-2"
        onClick={() => {
          props.onEdit(props.data);
          props.toggle();
        }}
      >
        Editar
      </button>
      <button
        id="delete"
        type="button"
        className="btn btn-outline-info btn-sm mr-sm-2"
        onClick={() => props.onDelete(props.data)}
      >
        Excluir
      </button>
      <button
        id="like"
        type="button"
        className="btn btn-outline-success btn-sm mr-sm-2"
        onClick={() => props.onVote({ id: props.data.id, option: "upVote" })}
      >
        Gostei
      </button>
      <button
        id="dislike"
        type="button"
        className="btn btn-outline-danger btn-sm"
        onClick={() => props.onVote({ id: props.data.id, option: "downVote" })}
      >
        Não Gostei
      </button>
    </div>
  );
};

CardBodyMain.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  author: PropTypes.string,
  timestamp: PropTypes.number,
  voteScore: PropTypes.number
};

export default CardBodyMain;
