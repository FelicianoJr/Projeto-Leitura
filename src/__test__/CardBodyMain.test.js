import React from "react";
import { shallow } from "enzyme";
import CardBodyMain from "../components/CardBodyMain";

const setup = (data, showPostButton) => {
  const actions = {
    toggle: jest.fn(),
    toggleComment: jest.fn()
  };
  const component = shallow(
    <CardBodyMain {...actions} data={data} showPostButton={showPostButton} />
  );

  return {
    component: component,
    actions: actions,
    divTitle: component.find(".card-title"),
    divBody: component.find(".card-text"),
    badgeScore: component.find(".badge"),
    b: component.find("b"),
  };
};

describe("<CardBodyMain/>", () => {

  const postProps = {
    title: "Udacity",
    body: "new situation",
    voteScore: 4,
    author: "Feliciano",
    timestamp: 142536
  };
  const commentProps = {
    body: "new situation",
    voteScore: 4,
    author: "Feliciano",
    timestamp: 142536
  };

  it("should render {title}", () => {
    const { divTitle, divBody, badgeScore, b } = setup(postProps);
    expect(divTitle.text()).toEqual("Udacity");
    expect(divBody.at(0).text()).toEqual("new situation");
    expect(badgeScore.text()).toEqual("4");
    expect(b.at(1).text()).toEqual("Feliciano");
    expect(b.at(2).text()).toEqual("142536");
  });

  it("renders a <FooterPost/>", () => {
    // const { footerPost } = setup(postProps,true);
    // expect(footerPost.props().data).toEqual({
    //   title: "Udacity",
    //   body: "new situation",
    //   voteScore: 4,
    //   author: "Feliciano",
    //   timestamp: 142536
    // });
  });

  it("renders a <FooterComment/>", () => {
    // const { footerComment } = setup(commentProps,false);
    // expect(footerComment.props().data).toEqual({
    //   body: "new situation",
    //   voteScore: 4,
    //   author: "Feliciano",
    //   timestamp: 142536
    // });
  });

  it("renders a cardBodyMain", () => {
    const { component } = setup(postProps);
    expect(component).toMatchSnapshot();
  });
});
