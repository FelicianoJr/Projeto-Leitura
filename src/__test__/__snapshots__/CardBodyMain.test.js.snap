// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBodyMain/> renders a cardBodyMain 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBodyMain
    data={
        Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 142536,
            "title": "Udacity",
            "voteScore": 4,
          }
    }
    showPostButton={undefined}
    toggle={[Function]}
    toggleComment={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          Udacity
</div>,
        <div
          className="card-text"
>
          new situation
</div>,
        <div
          className="card-text"
>
          &gt;
          <small>
                    <b>
                              Avaliação
                    </b>
          </small>
           
          <div
                    className="badge"
          >
                    4
          </div>
          <br />
          <small
                    className="text-muted"
          >
                     
                    Criado por 
                    <b>
                              Feliciano
                    </b>
                     em 
                    <b>
                              142536
                    </b>
          </small>
</div>,
        <Connect(FooterCardComment)
          data={
                    Object {
                              "author": "Feliciano",
                              "body": "new situation",
                              "timestamp": 142536,
                              "title": "Udacity",
                              "voteScore": 4,
                            }
          }
          toggleComment={[Function]}
/>,
      ],
      "className": "card-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Udacity",
          "className": "card-title",
        },
        "ref": null,
        "rendered": "Udacity",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "new situation",
          "className": "card-text",
        },
        "ref": null,
        "rendered": "new situation",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            ">",
            <small>
              <b>
                            Avaliação
              </b>
</small>,
            " ",
            <div
              className="badge"
>
              4
</div>,
            <br />,
            <small
              className="text-muted"
>
               
              Criado por 
              <b>
                            Feliciano
              </b>
               em 
              <b>
                            142536
              </b>
</small>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          ">",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                Avaliação
</b>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Avaliação",
              },
              "ref": null,
              "rendered": "Avaliação",
              "type": "b",
            },
            "type": "small",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "badge",
            },
            "ref": null,
            "rendered": 4,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Criado por ",
                <b>
                  Feliciano
</b>,
                " em ",
                <b>
                  142536
</b>,
              ],
              "className": "text-muted",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Criado por ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Feliciano",
                },
                "ref": null,
                "rendered": "Feliciano",
                "type": "b",
              },
              " em ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 142536,
                },
                "ref": null,
                "rendered": 142536,
                "type": "b",
              },
            ],
            "type": "small",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 142536,
            "title": "Udacity",
            "voteScore": 4,
          },
          "toggleComment": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            Udacity
</div>,
          <div
            className="card-text"
>
            new situation
</div>,
          <div
            className="card-text"
>
            &gt;
            <small>
                        <b>
                                    Avaliação
                        </b>
            </small>
             
            <div
                        className="badge"
            >
                        4
            </div>
            <br />
            <small
                        className="text-muted"
            >
                         
                        Criado por 
                        <b>
                                    Feliciano
                        </b>
                         em 
                        <b>
                                    142536
                        </b>
            </small>
</div>,
          <Connect(FooterCardComment)
            data={
                        Object {
                                    "author": "Feliciano",
                                    "body": "new situation",
                                    "timestamp": 142536,
                                    "title": "Udacity",
                                    "voteScore": 4,
                                  }
            }
            toggleComment={[Function]}
/>,
        ],
        "className": "card-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Udacity",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "Udacity",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "new situation",
            "className": "card-text",
          },
          "ref": null,
          "rendered": "new situation",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              ">",
              <small>
                <b>
                                Avaliação
                </b>
</small>,
              " ",
              <div
                className="badge"
>
                4
</div>,
              <br />,
              <small
                className="text-muted"
>
                 
                Criado por 
                <b>
                                Feliciano
                </b>
                 em 
                <b>
                                142536
                </b>
</small>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            ">",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  Avaliação
</b>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Avaliação",
                },
                "ref": null,
                "rendered": "Avaliação",
                "type": "b",
              },
              "type": "small",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "badge",
              },
              "ref": null,
              "rendered": 4,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Criado por ",
                  <b>
                    Feliciano
</b>,
                  " em ",
                  <b>
                    142536
</b>,
                ],
                "className": "text-muted",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Criado por ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Feliciano",
                  },
                  "ref": null,
                  "rendered": "Feliciano",
                  "type": "b",
                },
                " em ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 142536,
                  },
                  "ref": null,
                  "rendered": 142536,
                  "type": "b",
                },
              ],
              "type": "small",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "author": "Feliciano",
              "body": "new situation",
              "timestamp": 142536,
              "title": "Udacity",
              "voteScore": 4,
            },
            "toggleComment": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
